### ISSUE TITLE
NG 5.1 crashes when Android phone USB switches to Transfer Mode

### ISSUE THREAD

**Issue by Stargazzer** [htruong/crankshaft#379]
If you need support: Please post on https://www.reddit.com/r/crankshaft/ as this is a bug tracker, not a support forum.

Note: Reporting issues not based on official published builds will be closed without any comment.

- What phone do you have?
Umidigi Z2 Pro

- What OS version?
Android 8.1.0

- What crankshaft version?
5.1 (newest) with official 7" touch screen

- Steps to reproduce the bug, if you're planning to report a bug. Please indicate whether the bug is always repoducible.
Always happens, can not launch AA.

- Please provide any further information that you might find helpful if available.

Used Crankshaft/Android Auto on my way to work this morning and everything worked.

Took my pi/unit into the office to try the new update over internet option. Network cable not wifi.

There was an update so I did an update. If I remember it correctly it was the binaries that had an update.

Now the pi will not launch AA from my phone.  It sees/recognizes my phone and my phone charges but no AA.

On my phone I go to the USB settings and it is set for Transfer protocol.  But when i plug my phone into my pi it switches to charge only. 

On my phone if I tap the option and change it to File Transfer the NG 5.1 crashes and restarts.

On my phone in developer settings I have USB Debugging turned Off, USB Configuration is set to MTP Media Transfer Protocol.

I have also tried with USB Debugging ON but still the same crash.

No apps or Google things have been updated on my phone.  Again it worked on this mornings commute.

Any help would be great.

Thanks,

**Comment by abraha2d** [comment]
When the pi sees your phone, did you try pressing the USB button? I use the classic UI, and a USB button with the AA logo pops up on the right side. Sometimes it doesn't launch automatically, and I have to press that button manually to launch it.

My phone is set to charge only, and AA still works. Never tried it with transfer mode, will see if I can reproduce the crash.

**Comment by Stargazzer** [comment]
Hello Abraha2d,

Normal operation AA starts automatically when I plug in my phone into my pi running Crankshaft.
I honestly dont know if it changes to Charge Only under normal operation.  I only noticed it now because I was trying to troubleshoot and watched it when plugging in my phone.

Now when I plug my phone in, Crankshaft sees it but doesn't launch automatically.  So when I press the button to launch it nothing happens.  I get the notification that it is "Trying USB connect..." but nothing happens.  If I press the button again Crankshaft crashes and restarts.

Also something I have just noticed, might have been there before this issue but when I plug my phone in at the top left of the screen there is a green Lock icon.

Thanks for any help.
